### PR TITLE
First pass at #11130, allowing all available taxonomies to be queried…

### PR DESCRIPTION
**Pull request by kkoppenhaver**
… in WC REST API

### CODE REVIEW

File: includes/api/class-wc-rest-products-controller.php [hunk 1]
```diff
@@ -134,6 +134,31 @@ public function query_args( $args, $request ) {
...
+		*/
+
+		// Set up our temporary arguments to grab a product post.
+		$tmp_args = array(
+			'post_type' => 'products',
+			'posts_per_page' => 1,
+		);
+
+		// Grab the first product.
+		$tmp_product = get_posts( $args )[0];
```
Comment (kkoppenhaver) [inline includes/api/class-wc-rest-products-controller.php]: This is the line that failed Travis checks for PHP 5.2 and 5.3

`PHP Parse error:  syntax error, unexpected '['`

File: includes/api/class-wc-rest-products-controller.php [hunk 2]
```diff
@@ -141,12 +141,12 @@ public function query_args( $args, $request ) {
...
 		// Set up our temporary arguments to grab a product post.
 		$tmp_args = array(
-			'post_type' => 'products',
+			'post_type' => 'product',
 			'posts_per_page' => 1,
 		);
 
 		// Grab the first product.
-		$tmp_product = get_posts( $args )[0];
+		$tmp_product = get_posts( $tmp_args )[0];
```
Comment (kkoppenhaver) [inline includes/api/class-wc-rest-products-controller.php]: Still failing Travis on PHP 5.2 and 5.3

`Parse error: syntax error, unexpected '['`

**Comment by kkoppenhaver** [comment]
I'll be looking into why this is failing on PHP 5.2 and 5.3.  Will also be squashing commits when finally resolved.

**Comment by claudiosmweb** [comment]
@kkoppenhaver it's failing because `function()[0]` is a incorrect syntax for PHP 5.2.

**Comment by kkoppenhaver** [comment]
@claudiosmweb If that issue is fixed, do you think this is something with the potential to get merged given https://github.com/woothemes/woocommerce/issues/11130 ?

**Comment by claudiosmweb** [comment]
@kkoppenhaver this have a lot of problems, you don't need to use `get_posts()`, only `get_post()` should work, but variations have different taxonomies too and we'll not support custom taxonomies by default. You can do it using filters.

**Comment by kkoppenhaver** [comment]
@claudiosmweb Good to know, I'll take a look, thanks.

**Pull request closed without merging**